src/qt/bitcoin.cpp

```diff
@@ -486,6 +486,14 @@ int GuiMain(int argc, char* argv[])
     QApplication::setAttribute(Qt::AA_DontUseNativeMenuBar);
     QApplication::setAttribute(Qt::AA_DontCreateNativeWidgetSiblings);
     QApplication::setAttribute(Qt::AA_DontUseNativeDialogs);
+
+    // Add new safety native permission dialog for R/W actions.
+    static const QString write_permission = "android.permission.WRITE_EXTERNAL_STORAGE";
+    if (QtAndroid::checkPermission(write_permission) == QtAndroid::PermissionResult::Denied)
```

Comment:
Is the rationale for this to be able to write to SD card?

Reply:
Yes, we need to this for internal and removable volumes, such as an SD card, appear in the file system as part of external storage.

Reply:
@icota Thank you for review!
I checked the code again, there is no need for the Java method, and the last changes I made are enough on the C++ side.